### PR TITLE
please review this code setup

### PR DESCRIPTION
**Pull request by LeoMurphy7**

### CODE REVIEW

File: src/pages/login/login.jsx [hunk 1]
```diff
@@ -1,34 +1,79 @@
 import React, { useState } from "react";
 import { useAuth } from "../../utils/auth";
 import { useLocation, useNavigate } from "react-router";
+import { useFormik } from "formik";
+import * as Yup from "yup";
 
 export default function Login() {
-  const [user, setUser] = useState("");
+  // const [user, setUser] = useState({});
```
Comment (MoeThet1998) [inline src/pages/login/login.jsx:8]: if don't use remove it

File: src/pages/login/login.jsx [hunk 2]
```diff
@@ -1,34 +1,79 @@
...
   const auth = useAuth();
   const navigate = useNavigate();
   const location = useLocation();
-  const redirectPath = location.state?.path || "/" ;
+  const redirectPath = location.state?.path || "/";
```
Comment (MoeThet1998) [inline src/pages/login/login.jsx:12]: constant

File: src/pages/login/login.jsx [hunk 3]
```diff
@@ -1,34 +1,79 @@
...
 
-  const handleChange = (event) =>{
-    setUser(event.target.value);
-  }
+  // const handleChange = (event) => {
```
Comment (MoeThet1998) [inline src/pages/login/login.jsx:14]: if don't use remove it

File: src/pages/login/login.jsx [hunk 4]
```diff
@@ -1,34 +1,79 @@
...
+  //   setUser((prev)=> ({...prev, ["email"]: event.target.value }));
+  //   console.log('checking form input..',user)
+  // };
 
-  const handleLogin = () => {
-    auth.login(user);
+  const handleLogin = (value) => {
+    // auth.login(user);
```
Comment (MoeThet1998) [inline src/pages/login/login.jsx:20]: remove

File: src/pages/login/login.jsx [hunk 5]
```diff
@@ -1,34 +1,79 @@
...
+          <div className="mb-2 d-flex align-items-center">
+            <label htmlFor="password">Password: </label>
+            <input
+              className="form-control"
+              type="password"
+              id="password"
+              name="password"
+              {...formik.getFieldProps("password")}
+            />
+            {formik.touched.password && formik.errors.password ? (
```
Comment (MoeThet1998) [inline src/pages/login/login.jsx:69]: &&

File: src/pages/login/login.jsx [hunk 6]
```diff
@@ -1,34 +1,79 @@
...
+              Username:
+            </label>
+            <input
+              type="text"
+              id="email"
+              className="form-control"
+              onChange={formik.handleChange}
+              value={formik.values.email}
+            />
+            {formik.errors.email ? (
```
Comment (MoeThet1998) [inline src/pages/login/login.jsx:55]: &&

File: src/pages/login/login.jsx [hunk 7]
```diff
@@ -1,34 +1,79 @@
...
-    auth.login(user);
+  const handleLogin = (value) => {
+    // auth.login(user);
+    auth.login(value);
     navigate(redirectPath, { replace: true });
   };
 
+  const formik = useFormik({
+    initialValues: {
+      email: "",
```
Comment (MoeThet1998) [inline src/pages/login/login.jsx:27]: Hello

File: src/pages/login/login.jsx [hunk 8]
```diff
@@ -1,34 +1,79 @@
...
+            ) : null}
+          </div>
+
+          <div className="mb-2 d-flex align-items-center">
+            <label htmlFor="password">Password: </label>
+            <input
+              className="form-control"
+              type="password"
+              id="password"
+              name="password"
```
Comment (MoeThet1998) [inline src/pages/login/login.jsx:66]: const PASSWORD = "password"

File: src/pages/login/login.jsx [hunk 9]
```diff
@@ -1,34 +1,79 @@
...
   };
 
+  const formik = useFormik({
+    initialValues: {
+      email: "",
+      password: "",
+    },
+    validationSchema: Yup.object({
+      email: Yup.string().email("Invalid Email Address").required("Required"),
+      password: Yup.string().required("Reqired"),
```
Comment (MoeThet1998) [inline src/pages/login/login.jsx:32]: PASSWORD: Yup.string().required("Required")

File: src/pages/login/login.jsx (hunk 8 above)
Comment (MoeThet1998) [reply src/pages/login/login.jsx:66]: FieldName.js

const Login = {
EMAIL : 'email',
PASSWORD : "password"
}

File: src/pages/login/login.jsx (hunk 1 above)
Comment (LeoMurphy7) [reply src/pages/login/login.jsx:8]: done

File: src/pages/login/login.jsx (hunk 2 above)
Comment (LeoMurphy7) [reply src/pages/login/login.jsx:12]: done

File: src/pages/login/login.jsx (hunk 3 above)
Comment (LeoMurphy7) [reply src/pages/login/login.jsx:14]: done

File: src/pages/login/login.jsx (hunk 4 above)
Comment (LeoMurphy7) [reply src/pages/login/login.jsx:20]: done

File: src/pages/login/login.jsx (hunk 5 above)
Comment (LeoMurphy7) [reply src/pages/login/login.jsx:69]: done

File: src/pages/login/login.jsx (hunk 6 above)
Comment (LeoMurphy7) [reply src/pages/login/login.jsx:55]: done

**Pull request merged**